aioredlock/algorithm.py

```diff
@@ -119,7 +137,10 @@ def log(self):
 
             raise
 
-        return Lock(self, resource, lock_identifier, valid=True)
+        lock = Lock(self, resource, lock_identifier, valid=True)
+        self.watchdogs[lock.resource] = asyncio.ensure_future(self.__auto_extend(lock, self.auto_extend_count))
```

Comment:
Watchdog should not be created at all if `auto_extend_count` is 0.

Reply:
oh, yeah.u are right. I'll fix this

Reply:
Actually, even the watchdog is created when auto_extend_count us 0, it will be destroyed when self.__auto_entend is called first time. So this will have no side effects.